core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockDataMap.java

```diff
@@ -97,10 +98,15 @@
    * partition table and non transactional table
    */
   protected boolean isFilePathStored;
+  /**
+   * datamap model
+   */
+  protected BlockletDataMapModel blockletDataMapModel;
```

Comment:
Don't take complete `blockletDataMapModel` to class level, it wastes memory as we keep them in LRU . Just keep boolean isStandardCarbonTable in class level

Reply:
ok